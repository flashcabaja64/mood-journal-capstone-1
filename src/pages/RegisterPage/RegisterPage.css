h2.register-title {
  line-height: 1.2;
  text-align: center;
  font-size: 2.5rem;

  display: block;
  width: 100%;
  padding-top: 5%;
  padding-bottom: 4%;
}

.register_page {
  margin: 0 auto;
  min-width: 100vw;
  text-align: center;
  display: inline-block;
}