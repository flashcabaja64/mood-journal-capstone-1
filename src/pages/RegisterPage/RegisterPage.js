import React, { Component } from 'react';
import Register from '../../components/Register/Register';
import './RegisterPage.css'

export default class RegisterPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegisterSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {
    return(
      <div className="register_page">
        <h2 className="register-title">Start Journaling Now</h2>
        <Register
          onRegisterSuccess={this.handleRegisterSuccess}
        />
      </div>
    )
  }
}