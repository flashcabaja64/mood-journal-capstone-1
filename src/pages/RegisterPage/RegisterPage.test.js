import React from 'react';
import ReactDOM from 'react-dom';
import RegisterPage from './RegisterPage'
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<RegisterPage />, div)

  ReactDOM.unmountComponentAtNode(div)
})

it('renders UI as expected', () => {
  const tree = renderer
  .create(<RegisterPage />)
  .toJSON();
  expect(tree).toMatchSnapshot();
})