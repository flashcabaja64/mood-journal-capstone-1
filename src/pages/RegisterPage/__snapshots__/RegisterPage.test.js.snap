// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="register_page"
>
  <h2
    className="register-title"
  >
    Start Journaling Now
  </h2>
  <div
    className="main-register"
  >
    <div
      className="wrap-register"
    >
      <div
        className="form-wrap-register"
      >
        <form
          className="signup-form"
          onSubmit={[Function]}
        >
          <div
            className="register-input full_name"
          >
            <label
              htmlFor="full_name"
            >
              Full-Name
            </label>
            <input
              className="register-full_name"
              id="full_name"
              name="full_name"
              onChange={[Function]}
              placeholder="Full-Name"
              required={true}
              type="text"
            />
            <span
              className="focus-input"
            />
            <div
              className="error-message"
            >
              Input field cannot be empty
            </div>
          </div>
          <div
            className="register-input nickname"
          >
            <label
              htmlFor="nickname"
            >
              Nickname
            </label>
            <input
              className="register-nickname"
              id="nickname"
              name="nickname"
              onChange={[Function]}
              placeholder="Nickname"
              type="text"
            />
            <span
              className="focus-input"
            />
          </div>
          <div
            className="register-input user_name"
          >
            <label
              htmlFor="user_name"
            >
              Username
            </label>
            <input
              className="register-user_name"
              id="user_name"
              name="user_name"
              onChange={[Function]}
              placeholder="Username"
              required={true}
              type="text"
            />
            <span
              className="focus-input"
            />
            <div
              className="error-message"
            >
              Input field cannot be empty
            </div>
          </div>
          <div
            className="register-input password"
          >
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="register-password"
              id="password"
              maxLength="31"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <span
              className="focus-input"
            />
            <div
              className="error-message"
            >
              Password must be longer than 8 characters
            </div>
          </div>
          <div
            className="register-btn-container"
          >
            <button
              className="register-btn"
              disabled="Input field cannot be empty"
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
